### PR TITLE
Aw 83 - Lock Account after set amount of time

### PR DESCRIPTION
**Pull request by centrys-otniel**

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -136,6 +136,7 @@
         <javac
                 debuglevel="source,lines,vars"
                 includeantruntime="false"
+                debug="true"
```
Comment (centrys-otniel) [inline build.xml]: will remove this :D

Comment (centrys-cristian-ilca) [reply build.xml]: :+1:

File: src/main/java/org/aion/wallet/account/AccountManager.java [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+        return addressToAccount.keySet();
+    }
+
+    public AccountDTO getAccount(final String publicAddress) {
+        final String name = getStoredAccountName(publicAddress);
+        final String balance = BalanceUtils.formatBalance(balanceProvider.apply(publicAddress));
+        return new AccountDTO(name, publicAddress, balance, currencySupplier.get());
+    }
+
+    public void updateAccount(final AccountDTO account) {
```
Comment (centrys-cristian-ilca) [inline src/main/java/org/aion/wallet/account/AccountManager.java]: updateAccountName maybe?

File: src/main/java/org/aion/wallet/account/AccountManager.java [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+    private void handleSettingsChanged(final SettingsEvent event) {
+        if (SettingsEvent.Type.CHANGED.equals(event.getType())) {
+            final LightAppSettings settings = event.getSettings();
+            if (settings != null) {
+                lockTimeOut = settings.getUnlockTimeout();
+            }
+        }
+    }
+
+    private AccountDTO createAccountWithPrivateKey(final String address, final byte[] privateKeyBytes) {
```
Comment (centrys-cristian-ilca) [inline src/main/java/org/aion/wallet/account/AccountManager.java]: unchecked params?

File: src/main/java/org/aion/wallet/account/AccountManager.java [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+        final String name = getStoredAccountName(address);
+        final String balance = BalanceUtils.formatBalance(balanceProvider.apply(address));
+        AccountDTO account = new AccountDTO(name, address, balance, currencySupplier.get());
+        account.setPrivateKey(privateKeyBytes);
+        account.setActive(true);
+        addressToAccount.put(account.getPublicAddress(), account);
+        return account;
+    }
+
+    private void processAccountAdded(final AccountDTO account, final byte[] keystoreContent, final boolean isCreated) {
```
Comment (centrys-cristian-ilca) [inline src/main/java/org/aion/wallet/account/AccountManager.java]: unchecked params?

File: src/main/java/org/aion/wallet/events/RefreshEvent.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package org.aion.wallet.events;
+
+public class RefreshEvent extends AbstractEvent<RefreshEvent.Type> {
+
+    public static final String ID = "ui.data_refresh";
+
+    public RefreshEvent(Type eventType) {
```
Comment (centrys-cristian-ilca) [inline src/main/java/org/aion/wallet/events/RefreshEvent.java]: final?

File: src/main/java/org/aion/wallet/ui/components/SettingsController.java [hunk 6]
```diff
@@ -64,11 +73,28 @@ private void handleHeaderPaneButtonEvent(final HeaderPaneButtonEvent event) {
...
         settings = blockchainConnector.getSettings();
         protocol.setText(settings.getProtocol());
         address.setText(settings.getAddress());
         port.setText(settings.getPort());
-        notification.setText("");
+        timeout.setText(settings.getUnlockTimeout().toString().substring(2).toLowerCase());
+        displayNotification("", false);
+    }
+
+    private void displayNotification(final String message, boolean isError) {
```
Comment (centrys-cristian-ilca) [inline src/main/java/org/aion/wallet/ui/components/SettingsController.java]: final boolean isError

File: src/main/java/org/aion/wallet/ui/components/partials/PeerCountController.java [hunk 7]
```diff
@@ -35,10 +34,12 @@ protected final void refreshView(final RefreshEvent event) {
     }
 
     private void setPeerCount(int numberOfPeers) {
```
Comment (centrys-cristian-ilca) [inline src/main/java/org/aion/wallet/ui/components/partials/PeerCountController.java]: final

File: src/main/java/org/aion/wallet/ui/components/partials/PeerCountController.java [hunk 8]
```diff
@@ -35,10 +34,12 @@ protected final void refreshView(final RefreshEvent event) {
...
-        if(numberOfPeers == 1) {
-            peerCount.setText(numberOfPeers + " peer");
-            return;
+        final String description;
+        if (numberOfPeers == 1) {
```
Comment (centrys-cristian-ilca) [inline src/main/java/org/aion/wallet/ui/components/partials/PeerCountController.java]: check numberOfPeers before using?

File: src/main/java/org/aion/wallet/account/AccountManager.java (hunk 2 above)
Comment (centrys-otniel) [reply src/main/java/org/aion/wallet/account/AccountManager.java]: I was thinking that the method names should describe generic operations and the actual nature of the update would not be visible from outside, allowing us to hide the implementation's details

**Pull request merged**